pkg/limits/ingest_limits.go

```diff
@@ -389,62 +353,57 @@ func (s *IngestLimits) GetAssignedPartitions(_ context.Context, _ *logproto.GetA
 	return &resp, nil
 }
 
-// GetStreamUsage implements the logproto.IngestLimitsServer interface.
+// ExceedsLimits implements the logproto.IngestLimitsServer interface.
 // It returns the number of active streams for a tenant and the status of requested streams.
-func (s *IngestLimits) GetStreamUsage(_ context.Context, req *logproto.GetStreamUsageRequest) (*logproto.GetStreamUsageResponse, error) {
-	// Get the cutoff time for active streams
-	cutoff := time.Now().Add(-s.cfg.WindowSize).UnixNano()
-
-	// Calculate the rate window cutoff in nanoseconds
-	rateWindowCutoff := time.Now().Add(-s.cfg.RateWindow).UnixNano()
-
-	// Count total active streams for the tenant
-	// across all assigned partitions and record
-	// the streams that have been seen within the
-	// window
+func (s *IngestLimits) ExceedsLimits(_ context.Context, req *logproto.ExceedsLimitsRequest) (*logproto.ExceedsLimitsResponse, error) {
 	var (
-		activeStreams uint64
-		totalSize     uint64
+		lastSeenAt = time.Now()
```

Comment:
You might want to stub this with https://github.com/coder/quartz to make it easier to test? We do it in `pkg/limits/frontend/cache.go`, `pkg/limits/evict.go` and `pkg/limits/partition_manager.go`, you can use these as a reference.